dashboard/src/shared/Auth.ts

```diff
@@ -94,6 +94,21 @@ export class Auth {
     return r.status === 403 && (!r.data || !r.data.message);
   }
 
+  // isAnonymous returns true if the message includes "system:anonymous"
+  // in response.data or response.data.message
+  // the k8s api server nowadays defaults to allowing anonymous
+  // requests, so that rather than returning a 401, a 403 is returned if
+  // RBAC does not allow the anonymous user access.
+  public static isAnonymous(response: AxiosResponse): boolean {
+    return (
+      (response?.data?.message && response.data.message.includes("system:anonymous")) ||
+      (response?.data &&
+        response.data &&
+        typeof response.data === "string" &&
+        response.data.includes("system:anonymous"))
+    );
```

Comment:
this condition is getting a bit messy :) Here is an alternative:

```
const msg = get(response, "data.message") || get(response, "data");
return typeof msg === "string" && msg.includes("system:anonymous");
```

(being `get` lodash.get)

Reply:
Not aware of lodash.get. Much simpler now, thanks for the tip!